### PR TITLE
Rewrote case classes tour

### PR DESCRIPTION
**Pull request by travissarles**
I wanted to keep the concept of case classes simple before moving on to pattern matching. I know there are a few features that I didn't discuss like toString, hashcode, etc. I didn't think they were worth mentioning but feel free to disagree.

### CODE REVIEW

**Comment by travis032654** [comment]
@heathermiller ready to merge

File: tutorials/tour/_posts/2017-02-13-case-classes.md [hunk 1]
```diff
@@ -9,135 +9,49 @@ categories: tour
...
 
+## Defining a case class
+A minimal case class requires the keywords `case class`, an identifier, and a parameter list (which may be empty):
 ```tut
-abstract class Notification
-case class Email(sourceEmail: String, title: String, body: String) extends Notification
-case class SMS(sourceNumber: String, message: String) extends Notification
-case class VoiceRecording(contactName: String, link: String) extends Notification
-```
+case class Souvenir()
```
Comment (julienrf) [inline tutorials/tour/_posts/2017-02-13-case-classes.md]: I’m not sure this is a good example of case class. We generally use case objects when there are no fields. To me, the purpose of case classes is to aggregate information (in their fields).

File: tutorials/tour/_posts/2017-02-13-case-classes.md [hunk 2]
```diff
@@ -9,135 +9,49 @@ categories: tour
...
-val firstSms = SMS("12345", "Hello!")
-val secondSms = SMS("12345", "Hello!")
-
-if (firstSms == secondSms) {
-  println("They are equal!")
-}
-
-println("SMS is: " + firstSms)
+val message2 = Message("jorge@catalonia.es", "guillaume@quebec.ca", "Com va?")
+val message3 = Message("jorge@catalonia.es", "guillaume@quebec.ca", "Com va?")
```
Comment (julienrf) [inline tutorials/tour/_posts/2017-02-13-case-classes.md]: These examples are excellent!

File: tutorials/tour/_posts/2017-02-13-case-classes.md (hunk 1 above)
Comment (travis032654) [inline tutorials/tour/_posts/2017-02-13-case-classes.md]: OK I'll change it...

**Pull request merged**